dom/src/main/scala/fs2/dom/ResizeObserverEntry.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import org.scalajs.dom
+
+abstract class ResizeObserverEntry[F[_]] private[dom] {
```

Comment:
Perhaps something like this?

Reply:
I'll take a more detailed look later but this is exactly what I was imagining!! Thank you so much, really appreciate it :)

Reply:
Cool, I can add something similar for `dom.MutationRecord` tomorrow.